components/cards/ActionForm.tsx

```diff
@@ -201,42 +206,93 @@ export function ActionForm(props: ActionFormProps) {
+    const assetId = new AssetId({
+      chainId: `eip155:${NETWORK_ID}`,
+      assetName: {
+        namespace: "erc721",
+        reference: licenseAddress.toLowerCase(),
+      },
+      tokenId: licenseId
+        ? licenseId.toString()
+        : new BN(selectedParcelId.slice(2), "hex").toString(10),
+    });
+
+    if (licenseId) {
+      const mediaGallery: MediaObject[] = [];
+      const basicProfile = {
+        name: content.name ?? "",
+        url: content.url ?? "",
+      };
+
+      const mediaGalleryCid = await ipfs.dag.put(mediaGallery, {
+        storeCodec: "dag-cbor",
       });
 
-      const model = new DataModel({
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        ceramic: ceramic as any,
-        aliases: GeoWebModel,
+      const basicProfileCid = await ipfs.dag.put(basicProfile, {
+        storeCodec: "dag-cbor",
       });
 
-      const _assetContentManager = new AssetContentManager(
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        ceramic as any,
-        model,
-        ceramic.did?.capability.p.iss ?? "",
-        assetId
-      );
-
-      const _basicProfileStreamManager = new BasicProfileStreamManager(
-        _assetContentManager
-      );
-      await _basicProfileStreamManager.createOrUpdateStream(content);
-
-      setSelectedParcelId(`0x${new BN(parcelId.toString()).toString(16)}`);
-    } else if (basicProfileStreamManager) {
-      // Use existing BasicProfileStreamManager
-      await basicProfileStreamManager.createOrUpdateStream(content);
+      const parcelRoot = {
+        basicProfile: basicProfileCid,
+        mediaGallery: mediaGalleryCid,
+      };
+
+      const parcelRootCid = await ipfs.dag.put(parcelRoot, {
+        storeCodec: "dag-cbor",
+      });
+      const bytes = dagjson.encode(parcelRootCid);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const doc = await TileDocument.deterministic(ceramic as any, {
+        controllers: [ceramic.did?.parent ?? ""],
+        family: `geo-web-parcel`,
+        tags: [assetId.toString()],
+      });
+      await doc.update(json.decode(bytes));
+
+      setSelectedParcelId(`0x${new BN(licenseId.toString()).toString(16)}`);
+    } else {
+      try {
+        const ownerId = new AccountId(
+          AccountId.parse(ceramic.did?.parent.split("did:pkh:")[1] ?? "")
+        );
+        const web3Storage = new Web3Storage({
+          token: process.env.NEXT_PUBLIC_WEB3_STORAGE_TOKEN ?? "",
+          endpoint: new URL("https://api.web3.storage"),
+        });
+        const geoWebContent = new GeoWebContent({
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          ceramic: ceramic as any,
+          ipfs,
+          web3Storage,
+        });
+        const rootCid = await geoWebContent.raw.resolveRoot({
+          parcelId: assetId,
+          ownerId,
+        });
+        const newRoot = await geoWebContent.raw.putPath(
```

Comment:
The `/basicProfile` and `/mediaGallery` paths are actually `CID`s to separate nodes. Perhaps the library should detect whether a `CID` or node is passed in here, and handle both cases.

To be safer for now, you could do
```
let newRoot = await geoWebContent.raw.putPath(rootCid, "/basicProfile/name", content.name)
newRoot = await geoWebContent.raw.putPath(rootCid, "/basicProfile/url", content.url)
```

Reply:
I just published `@geo-web/content@0.3.0` that will now allow a node to be passed in the API and converted to a `CID`, as long as the parent schema is provided. So this just needs to add `{parentSchema: "ParcelRoot"}`

Reply:
Nice, `parentSchema` is now used every `putPath()` call.